Store.js

```diff
@@ -140,14 +150,109 @@ define(["dcl/dcl", "decor/Invalidating"], function (dcl, Invalidating) {
 						collection.on("delete", this._itemRemoved.bind(this));
 						collection.on("refresh", this._refreshHandler.bind(this));
 					}
-					return this.processCollection(collection);
+				} else {
+					for (var i = 0; i < this.store.length; i++) {
+						// affect the callback to the observe function if the item is observable
+						Observable.observe(this.store[i], this.observeCallback.bind(this));
+					}
+					collection = processQueryResult.call(this, this.store.filter(this._isQueried, this));
+					// affect the callback to the observe function if the array is observable
+					this._observeResult = ObservableArray.observe(this.store, this.observeCallback.bind(this));
 				}
+				return this.processCollection(collection);
 			} else {
 				this.initItems([]);
 			}
 		},
 
 		/**
+		 * Synchronously deliver change records of the Observe function for the Array
+		 * Do NOT deliver change records of the Observe function for Objects
+		 */
+		deliver: dcl.superCall(function (sup) {
```

Comment:
Please refer to https://github.com/polymer/observe-js#observable. My comment here is about how `deliver()` (and `discardChagnes()` should work (rather than about `Observable.observe()` handle). `deliver()` should deliver all pending change records (also refer to http://arv.github.io/ecmascript-object-observe/#DeliverChangeRecords). `discardChanges()` should throw away all pending change records.